hal/mbed_compat.c

```diff
@@ -90,4 +91,15 @@ MBED_WEAK void spi_get_capabilities(PinName ssel, bool slave, spi_capabilities_t
     }
 }
 
-#endif
\ No newline at end of file
+#endif
+
+#if DEVICE_RESET_REASON
+// To be re-implemented in the target layer if required
+MBED_WEAK void hal_reset_reason_get_capabilities(reset_reason_capabilities_t *cap)
+{
+    cap->reasons = 1 << RESET_REASON_PIN_RESET | 1 << RESET_REASON_SOFTWARE;
```

Comment:
just a suggestion, would be more readable if  `( )` around `(1 << RESET_REASON_PIN_RESET)` 

`(1 << RESET_REASON_PIN_RESET) | (1 << RESET_REASON_SOFTWARE)`

Reply:
Updated as requested.